src/main/webapp/intrigue-api/metacards/metacards.js

```diff
@@ -368,45 +370,44 @@ const createMetacard = async (parent, args, context) => {
 }
 
 const saveMetacard = async (parent, args, context) => {
-  const { id, attrs } = args
+  const { id, attributes } = args
   if (
-    Array.isArray(attrs.metacard_tags) &&
-    attrs.metacard_tags.includes('query-template')
+    Array.isArray(attributes.metacard_tags) &&
+    attributes.metacard_tags.includes('query-template')
   ) {
     return await saveQueryTemplate(parent, args, context)
   }
 
-  const { fetch, fromGraphqlName, toGraphqlName } = context
+  const [oldMetacard] = await metacardsById(
+    parent,
+    { ids: [id], ...args },
+    context
+  )
+  const [oldMetacardAttrs] = oldMetacard.attributes
+  const { catalog, fromGraphqlName, toGraphqlName } = context
 
-  const attributes = Object.keys(attrs).map(attribute => {
-    const value = attrs[attribute]
-    return {
-      attribute: fromGraphqlName(attribute),
-      values: Array.isArray(value) ? value : [value],
+  Object.keys(oldMetacardAttrs).forEach(attrs => {
+    if (attributes[attrs] === undefined) {
+      attributes[attrs] = oldMetacardAttrs[attrs]
```

Comment:
We shouldn't modify args in place. I used [setIn](https://immutable-js.github.io/immutable-js/docs/#/setIn) to get around this.

Reply:
attributes are the new attributes that need to be updated. oldMetacardAttrs are all the attributes that exist for said metacard. This is just adding all the necessary keys to `attributes`to make it a full metacard. But I'll look into `setIn`